### ISSUE TITLE
couldn't populate an assocition if model is used in a manytomany through association as junction table

### ISSUE THREAD

**Issue by atiertant** [balderdashy/waterline#1096]
couldn't populate an assocition if model is used in a manytomany through association as junction table.
look at this sample : 

    var Driver = Waterline.Collection.extend({
      identity: 'Driver',
      attributes: {
        "driverId": {
          "type": "integer",
          "primaryKey": true
        },
        "driverName": {
          "type": "string"
        },
        "taxis": {
          "collection": "Taxi",
          "via": "driver",
          "through": "ride"
        },
        "rides":{
          "collection": "ride",
          "via": "driver"
        }
      }
    });

    var Taxi = Waterline.Collection.extend({
      identity: 'Taxi',
      attributes: {
        "taxiId": {
          "type": "integer",
          "primaryKey": true
        },
        "taxiMatricule": {
          "type": "string"
        },
        "drivers": {
          "collection": "Driver",
          "via": "taxi",
          "through": "ride"
        }
      }
    });

    var Ride = Waterline.Collection.extend({
      identity: 'Ride',
      attributes: {
        "taxi": {
          "model": "Taxi"
        },
        "driver": {
          "model": "Driver"
        }
      }
    });
    
    Driver.findOne().where({driverId : 1}).populate('rides')

return this results:

    {
      rides: [ 
        { taxiId: 1,
          taxiMatricule: 'taxi_1',
          createdAt: '2015-07-21T11:49:21.908Z',
          updatedAt: '2015-07-21T11:49:21.908Z' 
        } 
      ],
      driverId: 1,
      driverName: 'driver 1',
      createdAt: '2015-07-21T11:49:21.913Z',
      updatedAt: '2015-07-21T11:49:21.913Z' 
    }

rides does not containt a list of rides but a list of taxis...

**Comment by devinivy** [comment]
I would certainly like this to work.  @dmarcelino what do you think about adding this to 0.11.x?  Something related might already be in there.  If anyone has time to write a failing test, that would be wonderful!

**Comment by dmarcelino** [comment]
I'm sorry if I missed something but isn't this issue essentially #705? As in through associations are currently not properly implemented and require hacks to work while #705 intends to officially support through associations?

**Comment by atiertant** [comment]
@dmarcelino here the problem is not in through association,but on other association using same model passed in through.

**Comment by dmarcelino** [comment]
I see now, `ride` is both a junction collection and also part of a one-to-many association and is actually the one-to-many association where things break. Yeap, it would be good if we can support this.

I do think that this change will depend on #705 though as #705 has the potential for making significant changes to how through associations are defined, but I'm speculating :)

**Comment by particlebanana** [comment]
Mentioned in #1133 but I think this may have something to do with the table having a join table flag on it.

**Comment by sailsbot** [comment]
Thanks for posting, @atiertant.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by sailsbot** [comment]
Thanks for posting, @atiertant.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
This appears to be active.  @particlebanana and @atiertant have made progress here, but I'm not sure that this should be closed yet.

**Comment by sailsbot** [comment]
Thanks for posting, @atiertant.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by particlebanana** [comment]
I believe this has been patched by the various M:M through pr's and commits that have been added. Closing unless shown otherwise.
